rules/legacy/src/Rector/Include_/AddTopIncludeRector.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Legacy\Rector\Include_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Include_;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Declare_;
+use PhpParser\Node\Stmt\DeclareDeclare;
+use PhpParser\Node\Stmt\GroupUse;
+use PhpParser\Node\Stmt\InlineHTML;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Node\Stmt\Nop;
+use PhpParser\Node\Stmt\Use_;
+use PhpParser\Node\Stmt\UseUse;
+use Rector\Core\Exception\Rector\InvalidRectorConfigurationException;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use ReflectionClass;
+
+/**
+ * @see https://github.com/rectorphp/rector/issues/3679
+ *
+ * @see \Rector\Legacy\Tests\Rector\Include_\AddTopIncludeRector\AddTopIncludeRectorTest
+ */
+final class AddTopIncludeRector extends AbstractRector
+{
+    /**
+     * @var bool
+     */
+    private $completed = false;
+
+    /**
+     * @var string
+     */
+    private $file = '';
+
+    /**
+     * @var string
+     */
+    private $lastFileProcessed = '';
+
+    /**
+     * @var int
+     */
+    private $type = 0;
+
+    /**
+     * @var string[]
+     */
+    private $classNodes = [
+        Declare_::class => true,
+        DeclareDeclare::class => true,
+        GroupUse::class => true,
+        InlineHTML::class => true,
+        Namespace_::class => true,
+        Use_::class => true,
+        UseUse::class => true,
+    ];
+
+    /**
+     * @var string[] to match
+     */
+    private $patterns = [];
+
+    /**
+     * @var array
+     */
+    private $configuration = [];
+
+    public function __construct(array $configuration = [])
+    {
+        $this->configuration = $configuration;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Adds an include file at the top of matching files but not class definitions', [
+            new CodeSample(
+                '<?php
+
+if (isset($_POST["csrf"])) {
+    processPost($_POST);
+}',
+                '<?php
+
+include "autoloader.php";
+
+if (isset($_POST["csrf"])) {
+    processPost($_POST);
+}',
```

Comment:
In multine-samples, there should be now/heredoc format used: 
https://github.com/rectorphp/rector/blob/965cef98e218163cb43049c1e5ef0b32fa635eef/rules/renaming/src/Rector/Class_/RenameClassRector.php#L55-L79

This prevents spacing issues and missing newlines